### PR TITLE
[TE] Self-Serve Alert Page V1

### PR DESCRIPTION
**Pull request by newsummit**
This is a working placeholder for the new Alert Anomaly Page (currently disabled in the flow). This page is part of a flow where anomaly 'replay' is triggered after alert creation and the user waits for notification of successful alert setup... after which the list of alerts will be loaded.

screen shot 2017-10-30 at 6 20 13 pm

Second commit : 
screen shot 2017-11-07 at 11 00 08 am

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+      const anomalyStats = [
+        {
+          title: 'Number of anomalies',
+          text: 'Estimated average number of anomalies per month',
+          value: total,
+        },
+        {
+          title: 'Response rate',
+          text: 'Percentage of anomalies that has a response',
+          value: '87.1%'
```
Comment (newsummit) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: Hard-coded for now... until the next phase (load summary of stats from server).

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+  /**
+   * Pings the job-info endpoint to check status of an ongoing replay job.
+   * If there is no progress after a set time, we display an error message.
+   * TODO: Set error message on timeout
+   * @method checkReplayStatus
+   * @param {Number} jobId - the id for the newly triggered replay job
+   * @return {fetch promise}
+   */
+  checkReplayStatus(jobId) {
+    const that = this;
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: let's stay away from saving `this` into another a value. 
The combination of `arrow functions` and passing this context in `Ember.run.later` should allow us to get rid of the `that` 
```
run.later(myContext, function() {
  // code here will execute within a RunLoop in about 500ms with this == myContext
}, 500);
```
https://emberjs.com/api/ember/2.15/namespaces/Ember.run/methods/later?anchor=later

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        <h2 class="te-alert-page-pending__title">Setting up your alert</h2>
+        <div class="te-alert-page-pending__loader"></div>
+        <p class="te-alert-page-pending__text">
+          This may take up to a few minutes <br/>
+          We will send you an email when it's done!
+        </p>
+      </div>
+    {{/if}}
+  </div>
+</section>
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/template.hbs]: lint: missing new line at EOF

File: thirdeye/thirdeye-frontend/app/styles/wrapper/styles.scss [hunk 4]
```diff
@@ -159,6 +159,182 @@ body {
...
+}
+
+.te-alert-page-cards {
+  @include group-clearfix;
+  font-family: $source-sans-pro;
+  margin-top: 20px;
+
+  &__card {
+    list-style: none;
+    float: left;
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/styles/wrapper/styles.scss]: using display flex might be easier here: https://codepen.io/anon/pen/BmNBXx

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js (hunk 2 above)
Comment (justYves) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: ^ This the only blocking comment.

Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: Ah, yes - I forgot to change that!

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+
+  /**
+   * Generate date range selection options if needed
+   * @method renderDate
+   * @param {Number} range - number of days (duration)
+   * @return {String}
+   */
+  renderDate: function(range) {
+    const newDate = moment().subtract(range, 'days').endOf('day').utc().format("DD MM YYY");
+    return `Last ${range} Days (${newDate} to Today)`;
```
[the remark this answers is not in the document]
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: Yes - I'll make a note to allow that.

File: thirdeye/thirdeye-frontend/app/styles/wrapper/styles.scss (hunk 4 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/styles/wrapper/styles.scss]: nice - thanks.

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+import { task, timeout } from 'ember-concurrency';
+
+export default Ember.Controller.extend({
+  /**
+   * Set up to receive prompt to trigger page mode change.
+   * When replay is received as true, it indicates that this is a
+   * newly created alert and replay is needed in order to display anomaly data.
+   */
+  queryParams: ['replay'],
+  replay: false,
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: Please add jsdocs comments for this variable, as per Data UI's coding standards wiki

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+
+  /**
+   * Standard legend settings for graph
+   */
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: Missing `@type`

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+    // TODO: enable single day range
+    const newDate = moment().subtract(range, 'days').endOf('day').utc().format("DD MM YYY");
+    return `Last ${range} Days (${newDate} to Today)`;
+  },
+
+  /**
+   * Array containing the types of anomaly responses supported
+   * @type {Array}
+   */
+  responseOptions: ['Not reviewed yet', 'True anomaly', 'False alarm', 'I don\'t know', 'Confirmed - New Trend'],
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/controller.js]: I think it's more readable if we have the variables defined at the top before methods and computed properties. Let me know what you think.

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/route.js [hunk 9]
```diff
@@ -0,0 +1,295 @@
+import Ember from 'ember';
+import fetch from 'fetch';
+import moment from 'moment';
+import { checkStatus, checkStatusPost } from 'thirdeye-frontend/helpers/utils';
+
+/**
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/route.js]: Can these be defined inside the route declaration, or is there a reason why these variables are defined outside? I might be missing something here.

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/route.js [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+      // Fetch anomaly metrics for this alert
+      .then((metrics) => {
+        Object.assign(model, { anomalyMetrics: metrics });
+      })
+      .catch((errors) => {
+        Object.assign(model, { loadError: true, loadErrorMsg: errors });
+      });
+  },
+
+  setupController(controller, model) {
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/route.js]: Documentation please :)

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/route.js [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+      filterData: filters,
+      anomalyData: anomalies,
+      rangeOptions: [durationString(model.durationWindow, model.baseStart)],
+      selectedRangeOption: durationString(model.durationWindow, model.baseStart)
+    });
+
+    controller.initialize(replay);
+  },
+
+  resetController(controller, isExiting, transition) {
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/route.js]: Documentation here also

File: thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js [hunk 12]
```diff
@@ -50,6 +50,11 @@ export default Ember.Controller.extend({
   },
 
   /**
+   * Change this to activate new alert anomaly page redirect
+   */
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js]: Missing `@type`

**Pull request merged**